### ISSUE TITLE
Definition tooltip is not dismissable without moving focus

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#3615]
## Environment
React Storybook

> Operating system
Windows 7

> Browser
Firefox and Chrome

## Detailed description
Although the tooltip appears on both hover and keyboard focus, it cannot be dismissed except by moving focus or the mouse hover.

> What did you expect to happen?

A requirement of WCAG is:
> Dismissible
A mechanism is available to dismiss the additional content without moving pointer hover or keyboard focus, unless the additional content communicates an input error or does not obscure or replace other content;

The normal implementation is that Esc should dismiss the tooltip content.

> What happened instead?

ESC did nothing. Not able to find any other way to dismiss it. 
> What WCAG 2.1 checkpoint does the issue violate?

[Content on hover or focus](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html)

## Steps to reproduce the issue

1. Launch the Definitino tooltip form React Storybook
2. Tab to put keyboard focus on definition, or hover with mouse
3. Attempt to dismiss without moving focus

## Additional information
![image](https://user-images.githubusercontent.com/14298245/62253519-ffdd0d80-b3aa-11e9-901b-9a79dcbec56b.png)
Screenshot of storybook with custom tooltip exposed

**Comment by emyarod** [comment]
currently our definition and icon tooltips are pure CSS. I believe to handle keyboard events we'll have to introduce JS and make some breaking changes. will research this further
